GVFS/GVFS.FunctionalTests/Tools/GVFSServiceProcess.cs

```diff
@@ -3,15 +3,37 @@
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
 using System.ServiceProcess;
 using System.Threading;
 
 namespace GVFS.FunctionalTests.Tools
 {
     public static class GVFSServiceProcess
     {
-        public const string TestServiceName = "Test.GVFS.Service";
-        private const string ServiceNameArgument = "--servicename=" + TestServiceName;
+        private static readonly string ServiceNameArgument = "--servicename=" + TestServiceName;
+
+        public static string TestServiceName
+        {
+            get
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    // Service name is used in the lookup of GVFS.Service communication pipes
+                    // by clients for IPC. Cusom pipe names can be specified as command line
+                    // args to the Service during its startup. On the Mac, GVFS.Service started
+                    // for testing is the same instance as the one that is installed by the
+                    // Mac installer. Also it is not as easy as in Windows to pass command line
+                    // args (that are specific to testing) to GVFS.Service (Service on Mac is a
+                    // Launch agent and needs its args to be specified in its launchd.plist
+                    // file). So on Mac - during tests GVFS.Service is started without any
+                    // customized pipe name for testing.

```

Comment:
> Service on Mac is a Launch agent and needs its args to be specified in its launchd.plist file

Could we solve this problem by having a second launchd.plist that is used for functional tests?  If that's the only thing stopping us from having a separate GVFS.Service for tests it seems like it would be worth adding.

Also, unrelated to the functional tests, where is the socket file located for GVFS.Service on the Mac?

Reply:
That's probably the way to go if we need to, but I'd like to see if we have need of this before we add the complexity.

Reply:
> I'd like to see if we have need of this before we add the complexity.

Agreed, we can defer this work until we enable the functional tests interact with `GVFS.Service` on the Mac.